src/github.com/couchbase/sync_gateway/rest/changes_api.go

```diff
@@ -188,15 +188,19 @@ func (h *handler) sendSimpleChanges(channels base.Set, options db.ChangesOptions
 		}
 
 		encoder := json.NewEncoder(h.response)
+
+		var forceClose bool = false
 	loop:
 		for {
 			select {
 			case entry, ok := <-feed:
 				if !ok {
+					forceClose = true
 					break loop // end of feed
 				}
 				if nil != entry {
 					if entry.Err != nil {
+						forceClose = true
```

Comment:
When the feed returns this type of error, I don't think it could be in the waiting state, could it?